tests/test_jobstores.py

```diff
@@ -159,7 +160,28 @@ class TestDjangoJobStore:
     See 'test_apscheduler_jobstore.py' for details
     """
 
-    pass
+    @pytest.mark.django_db(transaction=True)
+    def test_database_connection_recovery(self, jobstore, settings):
+        """Test that get_due_jobs() first closes unusable or obsolete
+        connections, like the start of a Django request."""
+        # Enable persistent connections to make sure the error causes the need
+        # to reconnect.
+        settings.MAX_CONN_AGE = 100000
+
+        # Precondition: simple case works.
+        db.connection.close()
+        jobstore.get_due_jobs(datetime(2016, 5, 3))
+
+        # Cause the connection to be errored.
+        db.connection.close()
+        with pytest.raises(db.OperationalError):
+            with db.connection.cursor() as cursor:
+                cursor.execute("INVALID SYNTAX ERROR")
+        assert db.connection.errors_occurred
```

Comment:
It seems like the only purpose of executing a query here that deliberately raises an error, and then checking `errors_occurred`, is to determine if `django.db.backends.base.BaseDatabaseWrapper.close_if_unusable_or_obsolete()` was called.

We could probably achieve the same more simply by mocking that method out and checking if it was `called`.